### PR TITLE
Add cache to contentversioninfo

### LINKED ISSUE #26931: Add Cache when getting Versionable info
### Parent Issue

_No response_

### Task

Due to the lack of caching, system performance is severely impacted. Implementing a caching mechanism for retrieving Versionable information is imperative.

### Proposed Objective

Code Maintenance

### Proposed Priority

Priority 2 - Important

### Acceptance Criteria

_No response_

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by erickgonzalez**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 846a7cb</samp>

This pull request adds a new feature of caching contentlet version infos by identifier to improve performance and reduce database queries. It modifies the `IdentifierCache` interface and its implementation, the `IdentifierCacheImpl` class, and adds new integration tests for them. It also optimizes the `VersionableFactoryImpl` class to use the new cache and a new query method.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/business/IdentifierCache.java [hunk 1]
```diff
@@ -70,4 +72,9 @@ public String get404Group() {
 	    return "Identifier404Cache";
 	}
 
+	public abstract void putContentVersionInfos(String identifier,
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/IdentifierCache.java:75]: doc

File: dotCMS/src/main/java/com/dotmarketing/business/VersionableFactoryImpl.java [hunk 2]
```diff
@@ -476,7 +452,23 @@ private List<ContentletVersionInfo> findContentletVersionInfos(final String iden
...
-		return findContentletVersionInfos(identifier, -1);
+		if(identifier==null){
+			throw new DotRuntimeException("identifier cannot be null");
+		}
+
+		List<ContentletVersionInfo> infos = icache.getContentVersionInfos(identifier);
+		if (infos != null) {
+			return infos;
+		}
+		synchronized (identifier.intern()) {
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotmarketing/business/VersionableFactoryImpl.java:463]: I think we need to address 

https://sonarqube.dotcms.site/project/issues?pullRequest=26968&resolved=false&severities=BLOCKER%2CCRITICAL%2CMAJOR%2CMINOR&types=BUG&open=AYxawmUXwfF26KX8_jpm&id=dotCMS_core_AYSbIemxK43eThAXTlt-

We should use the a Striped lock mechanism, something like this:

https://www.baeldung.com/java-lock-stripping#bd-4-concurrent-access-with-striped

**Review by wezell — CHANGES_REQUESTED** [review]
see comments

**Review by jdotcms — APPROVED** [review]

**Review by wezell — APPROVED** [review]
Looks good after Striped lock changes.

**Pull request merged**